VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionResponsesFragment.java

```diff
@@ -102,9 +102,14 @@ public void onDestroy() {
     public void onEventMainThread(DiscussionCommentPostedEvent event) {
         if (discussionThread.containsComment(event.getComment())) {
             discussionThread.incrementCommentCount();
-            courseDiscussionResponsesAdapter.setDiscussionThread(discussionThread);
+            if (event.getParent() == null) {
+                // We got a response
+                courseDiscussionResponsesAdapter.addNewResponse(event.getComment());
+            } else {
+                // We got a comment to a response
+                courseDiscussionResponsesAdapter.addNewComment(event.getParent());
+            }
```

Comment:
Also, instead of creating dedicated public methods for adding responses and comments in the adapter to be invoked from here, you could simply register for these events from the adapter itself, and perform the necessary checks and additions in there. Note that to be technically correct, the response would also need to be validated as the child of the adapter's thread.